### ISSUE TITLE
warning repeated infinitely

### ISSUE THREAD

**Issue by whohoho** [roburio/openvpn#28]
openvpn_client_lwt.exe: [WARNING] gethostbyname for localhost return an error: Not_found

**Comment by hannesm** [comment]
the code doing this is in `engine.ml`:
```OCaml
  | Resolving (idx, _, retry), `Resolve_failed ->
    next_or_fail idx retry >>| fun (state, action) ->
    { t with state = client state }, [], Some action
````

and `next_or_fail` using this function:
```OCaml
  and retry_exceeded r =
    match Config.get Connect_retry_max t.config with
    | `Times m -> m > r
    | `Unlimited -> false
```

with `Connect_retry_max` defaulting to `` `Unlimited``. now the question is whether this is intentional, I guess it is (we may want to check the original openvpn behaviour in such a setting), since at some point DNS resolution may work again (i.e. when your host gets back connectivity). Maybe there should be an exponential backoff instead of a linear one.

but the other issue here is that our Dns_client should treat `localhost` specially (and not forward to a remote resolver). There's some code in Dns_resolver (used e.g. by Dns_stub) to cope with this. Various zones are reserved and should not leave the host (or the local network). In the meantime, you can use an IP address :)

**Comment by hannesm** [comment]
thanks again for the report. this has been addressed in dns-client 4.4.1 (which is now merged into opam-repository) -- which now correctly resolves localhost to ::1 / 127.0.0.1. I'm closing this issue since nothing more needs to be done as far as I can tell. Please comment/reopen if you think we should do more about it.
